tests/Middleware/MiddlewareTestCase.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sentry\Tests\Middleware;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ServerRequestInterface;
+use Sentry\Configuration;
+use Sentry\Event;
+
+abstract class MiddlewareTestCase extends TestCase
+{
+    /**
+     * @param callable                    $middleware
+     * @param Event|null                  $event
+     * @param ServerRequestInterface|null $request
+     * @param \Exception|null             $exception
+     * @param array                       $payload
+     *
+     * @return Event The event returned by the middleware
+     */
+    protected function assertMiddlewareInvokesNext(callable $middleware, Event $event = null, ServerRequestInterface $request = null, \Exception $exception = null, array $payload = [])
+    {
+        $callbackInvoked = false;
+        $callback = function (Event $passedEvent, $passedRequest = null, $passedException = null) use (&$callbackInvoked) {
```

Comment:
This is intentional, to have the test failure on the subsequent assertion, not a type error here.

Reply:
Do we care about it? I mean, if a test fails because you're passing something that's not a `ServerRequestInterface` and the code crashes or because you're asserting that the type of the value is not the expected one is the same because we catch anyway the problem, don't we?

Reply:
Done.